### DISCUSSION TITLE
Example of next-urql with exchange-auth

### DISCUSSION THREAD

**Discussion by bytesofpie** [urql-graphql/urql#1542] [Q&A] [answered]
Hi! The urql docs are outstanding but I'm having a little trouble piecing together a working app with next-urql and exchange-auth. I'm using JWTs (auth and refresh) stored in local storage. In the functions I implement (`getAuth`, `addAuthToOperation`, etc), would I just check if I'm on the server and load the tokens from the context? I guess I'm having trouble wrapping my head around getting the tokens server side. I've tried combining examples from `next-urql`, `exchange-auth`, and `next-iron-session` with no luck. Is there an example out there already with a similar setup? If not I think this would be a great example to have as well once I figure it out.

EDIT: I should mention the backend api is in elixir so it's external and has a login and refresh mutation to supply tokens. Would I need to use cookies instead of localStorage in this case?

**Comment by JoviDeCroock** [comment, marked as answer]
If you require authentication on the server this will need to take the format of a cookie, this can be done through a library like [`nookies`](https://github.com/maticzav/nookies) where it will take care for you whether to use `set-cookie` or analyze `document.cookie`. There isn't really an example out there with cookies currently but the idea behind it is very similar to the one for localStorage/... only that you'll set/get the cookies from ctx/document.cookie.

**Reply by kitten** [reply]
It's also worth saying that your assumption is right; you don't have to use `@urql/exchange-auth` on the server-side. You can use `fetchOptions` on the server, since there's no need to add any auth-refresh logic.

**Reply by JoviDeCroock** [reply]
Well for a first visit there is a possibility that the user doesn't have a token yet where we could have to fetch a token. Unless the server-side renders use some kind off common token
